### PR TITLE
Add NSP API to multiple services under Microsoft.Insights RP

### PR DESCRIPTION
**Pull request by akshitgoyal**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary>Click here to see the details of Step 1, Breaking Changes review</summary>

If you are in purview of Step 1 of the diagram, follow the [Breaking Changes review process].  
**IMPORTANT!** This applies even if you believe your PR was mislabeled, for any reason, including tool failure.
</details>

<details> 
<summary>Click here to see the details of Step 2, ARM review</summary>

See https://aka.ms/azsdk/pr-arm-review.
</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[Breaking Changes review process]: https://aka.ms/brch

### CODE REVIEW

**Comment by raosuhas** [comment]
YOu have a large number of warnings in the LintDiff. Could you please fix these : https://github.com/Azure/azure-rest-api-specs/pull/28680/checks?check_run_id=23901771716

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/scheduledQueryRule_NetworkSecurityPerimeter_API.json [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the config.",
+            "schema": {
+              "$ref": "#/definitions/NetworkSecurityPerimeterConfiguration"
+            }
+          },
+          "default": {
+            "description": "BadRequest",
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/scheduledQueryRule_NetworkSecurityPerimeter_API.json:114]: >BadRequest [](http://example.com/codeflow?start=28&length=10)

should be generic enough for all types of errors ? Here and all through the PR

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+            "$ref": "#/parameters/ActionGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/configurationName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:158]: >responses [](http://example.com/codeflow?start=9&length=9)

will there be a case of post responding with 200, meaning that the reconcile on the configurationName is completed or when the Location header responds with a successful terminal state ?

here and in other files having POST calls.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+          "202": {
+            "description": "Request to reconcile the association accepted.",
+            "headers": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:161]: consider adding a retry-after?

here and in other files having POST calls.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+        }
+      }
+    },
+    "NSPConfigAssociation": {
+      "description": "Information about resource association",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:295]: any regex patterns , min, max length that can be added ? 

here and other places having string names

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+        "accessMode": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:296]: >accessMode [](http://example.com/codeflow?start=9&length=10)

are there any set of values which can go as enum ?

here and other places  in the PR

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+        }
+      }
+    },
+    "NSPConfigNetworkSecurityPerimeterRule": {
+      "description": "NetworkSecurityPerimeter related information",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:307]: is this an arm id ? if so, "format": "arm-id" can be added 

here and other places in the PR

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+        "perimeterGuid": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:310]: name says guid, but value is string . Please make it consistent.

here and other places in the PR

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "accessMode": {
+          "type": "string"
+        }
+      }
+    },
+    "NSPConfigNetworkSecurityPerimeterRule": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:301]: >NSPConfigNetworkSecurityPerimeterRule [](http://example.com/codeflow?start=5&length=37)

NSPConfigNetworkSecurityPerimeterRule and NSPConfigPerimeter are having same body. Can they be merged?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 9]
```diff
@@ -0,0 +1,419 @@
...
+        }
+      }
+    },
+    "NSPConfigPerimeter": {
+      "description": "NetworkSecurityPerimeter related information",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:326]: is this an arm id ? if so, "format": "arm-id" can be added

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 10]
```diff
@@ -0,0 +1,419 @@
...
+        "perimeterGuid": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:329]: name says guid, but value is string . Please make it consistent.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 11]
```diff
@@ -0,0 +1,419 @@
...
+        }
+      }
+    },
+    "NSPConfigProfile": {
+      "description": "Network Security Perimeter profile",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:345]: any regex patterns , min, max length that can be added ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+        }
+      }
+    },
+    "NSPProvisioningIssue": {
+      "description": "Describes provisioning issue for given NetworkSecurityPerimeterConfiguration",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:364]: any regex patterns , min, max length that can be added ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 13]
```diff
@@ -0,0 +1,419 @@
...
+    "NSPProvisioningIssueProperties": {
+      "description": "Properties of provisioning issue",
+      "type": "object",
+      "properties": {
+        "issueType": {
+          "type": "string"
+        },
+        "severity": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:379]: are there any set of values which can go as enum ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 14]
```diff
@@ -0,0 +1,419 @@
...
+        }
+      }
+    },
+    "NSPProvisioningIssueProperties": {
+      "description": "Properties of provisioning issue",
+      "type": "object",
+      "properties": {
+        "issueType": {
+          "type": "string"
+        },
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:376]: are there any set of values which can go as enum ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/scheduledQueryRule_NetworkSecurityPerimeter_API.json [hunk 15]
```diff
@@ -0,0 +1,417 @@
...
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationProperties": {
+      "description": "The properties of an NSP config.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/scheduledQueryRule_NetworkSecurityPerimeter_API.json:219]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

Enum ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json [hunk 16]
```diff
@@ -0,0 +1,429 @@
...
+            "$ref": "#/parameters/DataCollectionEndpointNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/configurationName"
+          }
+        ],
+        "responses": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json:145]: same as in actiongroups json file

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json [hunk 17]
```diff
@@ -0,0 +1,429 @@
...
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfigurationProperties": {
+      "description": "The properties of an NSP config.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json:213]: >provisioningState [](http://example.com/codeflow?start=9&length=17)

enum?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json [hunk 18]
```diff
@@ -0,0 +1,429 @@
...
+        "properties": {
+          "$ref": "#/definitions/NSPConfigAccessRuleProperties"
+        }
+      }
+    },
+    "NSPConfigAccessRuleProperties": {
+      "description": "Properties of Access Rule",
+      "type": "object",
+      "properties": {
+        "direction": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json:252]: >direction [](http://example.com/codeflow?start=9&length=9)

enum ?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json [hunk 19]
```diff
@@ -0,0 +1,429 @@
...
+          "type": "string"
+        },
+        "addressPrefixes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "fullyQualifiedDomainNames": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json:261]: >fullyQualifiedDomainNames [](http://example.com/codeflow?start=9&length=25)

arm-id ?

**Comment by ramoka178** [comment]
Could you please comment on the active comments + fix the merge conflict

**Comment by akshitgoyal** [comment]
> Could you please comment on the active comments + fix the merge conflict

We cannot make changes to these models as these are same for all the services and are based on guidance from networking team. I have linked swagger for two other services that have already gone through arm review and have the same models.
https://github.com/Azure/azure-rest-api-specs/blob/cf230c94500f8ee044acc68cb7bab37eace5671d/specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/NetworkSecurityPerimeterConfigurations.json

https://github.com/Azure/azure-rest-api-specs/blob/cf230c94500f8ee044acc68cb7bab37eace5671d/specification/storage/resource-manager/Microsoft.Storage/stable/2023-04-01/networkSecurityPerimeter.json

**Comment by ramoka178** [comment]
Please fix the merge conflict and place PR back in queue

**Comment by akshitgoyal** [comment]
> Please fix the merge conflict and place PR back in queue

resolved

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json [hunk 20]
```diff
@@ -0,0 +1,464 @@
...
+          "description": "FQDN for outbound rules",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "description": "Fully qualified domain name",
+            "type": "string"
+          }
+        },
+        "subscriptions": {
```
Comment (TimLovellSmith) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json:291]: >subscriptions [](http://example.com/codeflow?start=9&length=13)

description?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json [hunk 21]
```diff
@@ -0,0 +1,464 @@
...
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "networkSecurityPerimeters": {
```
Comment (TimLovellSmith) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/dataCollectionEndpoints_NetworkSecurityPerimeter_API.json:297]: >networkSecurityPerimeters [](http://example.com/codeflow?start=9&length=25)

description?

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json [hunk 22]
```diff
@@ -0,0 +1,454 @@
...
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "NetworkSecurityPerimeterConfiguration": {
```
Comment (TimLovellSmith) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:182]: >NetworkSecurityPerimeterConfiguration [](http://example.com/codeflow?start=5&length=37)

do you need to have multiple definitions of all these NetworkSecurityPerimeterConfiguration types? 
I think it'd be a great idea to deduplicate them, have them all in one networksecurityperimeter.json file, and share them, if you can

Comment (TimLovellSmith) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2021-10-01/actionGroups_NetworkSecurityPerimeter_API.json:182]: You should also compare your definitions with these, I think they are better documented:

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/botservice/resource-manager/Microsoft.BotService/preview/2023-09-15-preview/networksecurityperimeter.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/botservice/resource-manager/Microsoft.BotService/preview/2023-09-15-preview/networksecurityperimeter.json)

**Comment by TimLovellSmith** [comment]
@akshitgoyal I know this was technically ARMSignedOff already, and you should be unblocked for manifest deployment etc at this point, but I've been reviewing network security perimeter definitions for *several* RPs this week and its getting quite frustrating how much variety there is. I'd really appreciate it if you can help with this problem bring your API and docs closer to every other RP's before we actually merge this.

**Comment by akshitgoyal** [comment]
> @akshitgoyal I know this was technically ARMSignedOff already, which is good - you are now unblocked for RP manifest deployment etc at this point, I don't want to change that - but... there's one more thing before we merge this I've been reviewing network security perimeter definitions for _several_ RPs this week and its getting quite concerning how much unnecessary variety there is. I'd really appreciate it if you can help with this problem bring your API and docs closer to every other RP's (API consistency) before we actually merge this.

@TimLovellSmith we were given this RP's API (link attached) as a reference by the NSP team itself so we followed this: https://github.com/Azure/azure-rest-api-specs/blob/1fe785dc9b49c48088a9d11db023eee07dd168e7/specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/NetworkSecurityPerimeterConfigurations.json#L19

**Comment by TimLovellSmith** [comment]
@akshitgoyal I'm working on adding NetworkSecurityPerimeters to common types, as something like this is clearly urgently needed! Please help me review my PR.

https://github.com/Azure/azure-rest-api-specs/pull/28958

**Comment by akshitgoyal** [comment]
> @akshitgoyal I'm working on adding NetworkSecurityPerimeters to common types, as something like this is clearly urgently needed! Please help me review my PR.
> 
> #28958

@TimLovellSmith I started an email thread with Networking team for this.

**Review by TimLovellSmith — CHANGES_REQUESTED** [review]
Marking this PR as need to refactor to refer to NSP definitions from

https://github.com/Azure/azure-rest-api-specs/pull/28958

before merging.

To be clear the intent of this feedback is not to say that ARM is not signed off (ARM is signed off), nor is it to block manifest deployments (they should be unblocked)

**Comment by akshitgoyal** [comment]
> Marking this PR as need to refactor to refer to NSP definitions from
> 
> #28958
> 
> before merging.
> 
> To be clear the intent of this feedback is not to say that ARM is not signed off (ARM is signed off), nor is it to block manifest deployments (they should be unblocked)

I will refactor the code as soon as your PR is merged. Thanks!

**Comment by mentat9** [comment]
@akshitgoyal - Please fix Automated merging requirements and resolve merge conflicts. Afterward, re-apply the MergeRequested label.

**Comment by TimLovellSmith** [comment]
@akshitgoyal Please refactor by using references to the [new] common types NSP definitions, letting me know of any blockers

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/networksecurityperimeter.json

**Comment by akshitgoyal** [comment]
> @akshitgoyal Please refactor by using references to the [new] common types NSP definitions, letting me know of any blockers
> 
> https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/networksecurityperimeter.json

I have refactored the code to use the new types.

**Comment by akshitgoyal** [comment]
@TimLovellSmith I had to add parameter definition for NetworkSecurityPerimeterConfigurationNameParameter because the current common types don't have a "pattern" restriction and the CI checks are failing because of that. I think the common types definition needs to be updated to account for this. LintDiff will fail without that. 

if these changes look okay, can you approve them, so we can get this merged? Thanks. Can you also run `azp run`? For some reason, some CI checks are failing because my account has authorization issues.

**Comment by TimLovellSmith** [comment]
> had to add parameter definition for NetworkSecurityPerimeterConfigurationNameParameter because the current common types don't have a "pattern" restriction

Thats fine for now thanks!

**Review by TimLovellSmith — APPROVED** [review]

**Pull request merged**